### PR TITLE
Separate `java-runtime` module

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR extracts a `java-runtime` module so that projects that depend on `base` too can benefit from having a smaller JAR.

### CODE REVIEW

**Review by armiol — CHANGES_REQUESTED** [review]
@alexander-yevsyukov please see my comments. Looks IDEA wants to contribute as well :)

**Review by dmdashenkov — COMMENTED** [review]
LGTM on my part. I will, however, make one or two changes to `config` due to some oddities I've noticed in the `buildSrc` code.

File: java-runtime/src/main/java/io/spine/validate/option/RangeDecl.java [hunk 1]
```diff
@@ -94,11 +94,11 @@ static Range<ComparableNumber> compile(String rangeOptionValue, FieldDeclaration
     private RangeDecl(String rangeOptionValue, FieldDeclaration field) {
         var rangeMatcher = NUMBER_RANGE.matcher(rangeOptionValue.trim());
         checkState(rangeMatcher.matches(),
-                   "Malformed range `%s` on the field `%s`. " +
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply java-runtime/src/main/java/io/spine/validate/option/RangeDecl.java:97]: I'm on it...

**Comment by alexander-yevsyukov** [comment]
> LGTM on my part. I will, however, make one or two changes to `config` due to some oddities I've noticed in the `buildSrc` code.

@dmdashenkov, I've updated `config` already. What oddities you mean in particular, so we don't miss them addressed?

**Comment by alexander-yevsyukov** [comment]
> @alexander-yevsyukov, the `val about = ""` trick caught my eye, but I now see that it's intentional, not a mistake like I thought first.

@dmdashenkov, it's style of @yevhenii-nadtochii. I do it like this:

```kotlin
private const val ABOUT =""
```
This way there's no potential conflict with other `about`s. And it's more bold, like of one of brave's penguin.

**Pull request merged**